src/modules/Board/components/Board.tsx

```diff
@@ -0,0 +1,114 @@
+import AddIcon from '@mui/icons-material/Add';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { Container, Stack, Typography, Button } from '@mui/material';
+import { lightBlue } from '@mui/material/colors';
+import { useState, useEffect } from 'react';
+import { getBoardInfo } from '@/data/board';
+import { getStatuses } from '@/data/statuses';
+import { getTasks } from '@/data/tasks';
+import { getUsers } from '@/data/user';
+import { useDialogState } from '@/hooks/useDialogState';
+import { Task } from '@/types/task';
+import { useTasks, useUserData, useUserNames } from '../hooks';
+import { List } from './List';
+import { TaskFormDialog } from './TaskFormDialog';
+import { TaskInfoDialog } from './TaskInfoDialog';
+
+export function Board() {
+  const statuses = getStatuses();
+  const board = getBoardInfo();
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [tasks, setTasks] = useTasks(getTasks());
+  const [taskId, setTaskId, taskFormDialog] = useDialogState<string>('');
+  const [, , taskInfoDialog] = useDialogState<string>('');
+  const [users, setUsers] = useUserData([]);
+  const [userNames, setUserNames] = useUserNames({});
+  const [isCreate, setCreate] = useState<boolean>(false);
+  const [currentTask, setCurrentTask] = useState<Task>();
+
+  useEffect(() => {
+    const usersPromise = getUsers();
+    usersPromise.then((result) => {
+      const usersIdToName: { [id: string]: string } = {};
+      setUsers(result);
+      result.forEach((user) => {
+        usersIdToName[user.id] = user.name;
+      });
+      setUserNames(usersIdToName);
+    });
+  }, [setUserNames, setUsers]);
+
+  useEffect(() => {
+    setCurrentTask(tasks.find((task) => task.id === taskId));
```

Comment:
I think it's redundant to create two states for taskId and for currentTask. You can just store the task in data of  taskFormDialog.
I meant this implementation

```typescript
const [currentTask, setCurrentTask, taskFormDialog] = useDialogState<Task | undefined>(undefined);
```

and such implementation of onTaskClick
```typescript
  function onTaskClick(task: Task) {
    setCreate(false);
    setCurrentTask(task);
    taskInfoDialog.open();
  }
```